lightning/src/util/string.rs

```diff
@@ -40,3 +44,26 @@ mod tests {
 		);
 	}
 }
+
+/// Struct to `Display` fields in a safe way using `PrintableString`
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct SanitizedString(pub String);
+
+impl Writeable for SanitizedString {
+	fn write<W: Writer>(&self, w: &mut W) -> Result<(), io::Error> {
+		self.0.write(w)
+	}
+}
+
+impl Readable for SanitizedString {
+	fn read<R: Read>(r: &mut R) -> Result<Self, msgs::DecodeError> {
+		let s: String = Readable::read(r)?;
+		Ok(Self(s))
+	}
+}
+
+impl fmt::Display for SanitizedString {
+	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+		PrintableString(&self.0).fmt(f)
+	}
+}
```

Comment:
Could you move all this before `PrintableString`? We typically don't interleave code with test code.